ratis-shell/src/main/java/org/apache/ratis/shell/RetryUtil.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.shell;
+
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Retry related utilities function.
+ */
+public final class RetryUtil {
+
+  /**
+   * Execute a given function with input parameter from member of list.
+   *
+   * @param list the input parameters
+   * @param function the function to be executed
+   * @param <T> parameter type
+   * @param <K> return value type
+   * @return the function return value
+   */
+  public static <T, K> K run(List<T> list, Function<T, K> function) {
```

Comment:
Use org.apache.ratis.util.function.CheckedFunction so that the function can throw a checked exception.

Reply:
Thanks for show me this `CheckedFunction`.